independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/startup/RuntimeResourceDeployment.java

```diff
@@ -418,9 +427,9 @@ public RuntimeResource buildResourceMethod(ResourceClass clazz,
         abortHandlingChain.add(ExceptionHandler.INSTANCE);
         abortHandlingChain.add(ResponseHandler.NO_CUSTOMIZER_INSTANCE);
         abortHandlingChain.addAll(responseFilterHandlers);
-
         abortHandlingChain.add(responseWriterHandler);
-        handlers.add(0, new AbortChainHandler(abortHandlingChain.toArray(EMPTY_REST_HANDLER_ARRAY)));
+
+        handlers.setFirst(new AbortChainHandler(abortHandlingChain.toArray(EMPTY_REST_HANDLER_ARRAY)));
```

Comment:
Can't we just add this right at the start and avoid this custom list?

Reply:
That was my first thought, but it's not easy because that element needs handlers that are created further down the code, so it would require a lot of reorganizing